account_invoice_default_account_date/models/account_invoice.py

```diff
@@ -0,0 +1,20 @@
+from datetime import date
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def write(self, vals):
+        # vendor bills
+        if (
+            not self.date
+            and not vals.get("date")
```

Comment:
The python way: `and "date" not in vals`

Reply:
done